app/assets/javascripts/discourse/app/controllers/user-activity-bookmarks-with-reminders.js

```diff
@@ -38,69 +35,72 @@ export default Controller.extend({
+      }
+    });
+  },
 
-    if (response && response.no_results_help) {
-      this.set("noResultsHelp", response.no_results_help);
+  @action
+  editBookmark(bookmark) {
+    let controller = showModal("bookmark", {
+      model: {
+        postId: bookmark.post_id,
+        id: bookmark.id,
+        reminderAt: bookmark.reminder_at,
+        name: bookmark.name
+      },
+      title: "post.bookmarks.edit",
+      modalClass: "bookmark-with-reminder"
+    });
+    controller.setProperties({
+      afterSave: () => this.loadItems()
+    });
+  },
+
+  @action
+  loadMore() {
+    if (this.loadingMore) {
+      return Promise.resolve();
     }
 
-    this.model.more_bookmarks_url = response.more_bookmarks_url;
+    this.set("loadingMore", true);
 
-    if (response && response.bookmarks) {
-      let bookmarks = [];
-      response.bookmarks.forEach(bookmark => {
-        bookmarks.push(Bookmark.create(bookmark));
-      });
-      this.content.pushObjects(bookmarks);
-    }
+    return this.model
+      .loadMore()
+      .then(response => this._processLoadResponse(response))
+      .catch(() => this._bookmarksListDenied())
+      .finally(() => this.set("loadingMore", false));
   },
 
-  actions: {
-    removeBookmark(bookmark) {
-      bootbox.confirm(I18n.t("bookmarks.confirm_delete"), result => {
-        if (result) {
-          return bookmark.destroy().then(() => this.loadItems());
-        }
-      });
-    },
+  _bookmarksListDenied() {
+    this.set("noResultsHelp", I18n.t("bookmarks.list_permission_denied"));
+  },
 
-    editBookmark(bookmark) {
-      let controller = showModal("bookmark", {
-        model: {
-          postId: bookmark.post_id,
-          id: bookmark.id,
-          reminderAt: bookmark.reminder_at,
-          name: bookmark.name
-        },
-        title: "post.bookmarks.edit",
-        modalClass: "bookmark-with-reminder"
-      });
-      controller.setProperties({
-        afterSave: () => this.loadItems()
-      });
-    },
+  _processLoadResponse(response) {
+    if (!response) {
+      this._bookmarksListDenied();
+      return;
+    }
 
-    loadMore() {
-      if (this.loadingMore) {
-        return Promise.resolve();
-      }
-      this.set("loadingMore", true);
+    if (response.no_results_help) {
+      this.set("noResultsHelp", response.no_results_help);
+      return;
+    }
 
-      return this.model
-        .loadMore()
-        .then(response => this.processLoadResponse(response))
-        .catch(() => {
-          this.set("noResultsHelp", I18n.t("bookmarks.list_permission_denied"));
-        })
-        .finally(() =>
-          this.setProperties({
-            loadingMore: false
-          })
-        );
+    response = response.user_bookmark_list;
+    this.model.more_bookmarks_url = response.more_bookmarks_url;
+
+    if (response.bookmarks) {
+      this.content.pushObjects(
+        response.bookmarks.map(bookmark => Bookmark.create(bookmark))
```

Comment:
I find this kind of operation is much better suited for a model than a controller. Is there a reason we can't do the transformation in the model?